core/model/job/step/gotolocation/gotolocation.go

```diff
@@ -0,0 +1,105 @@
+package gotolocation
+
+import (
+	"context"
+	"time"
+)
+
+type State int
+type Event int
+
+const (
+	New State = iota
+	Moving
+	Near
+	Arrived
+	Cancelled
+)
+
+func (s State) String() string {
+	return [...]string{"New", "Moving", "Near", "Arrived", "Cancelled"}[s]
+}
+
+const (
+	MoveEvent Event = iota
+	NearEvent
+	ArrivedEvent
+	CancelledEvent
+)
+
+func (e Event) String() string {
+	return [...]string{"MoveEvent", "NearEvent", "ArrivedEvent", "CancelledEvent"}[e]
+}
+
+type GoToLocation struct {
+	DriverID             string  `json:",string"`
```

Comment:
What does `json:",string"` mean here?

Reply:
GoToLocation struct keeps state of step execution. This state is stored in redis in a HSET. Struct is marshalled to map[string]string because redis works with strings or binary representations of strings. Golang json marshal is used to serialise struct to map. these    `json:",string"` are hints for json marshaller.